prow/cmd/deck/static/pr/pr.ts

```diff
@@ -387,8 +387,10 @@ function getFullPRContext(builds: Job[], contexts: Context[]): UnifiedContext[]
  */
 function loadPrStatus(prData: UserData): void {
     const tideQueries: TideQuery[] = [];
-    for (const query of tideData.TideQueries) {
-        tideQueries.push(new TideQuery(query));
+    if (tideData.TideQueries != null) {
```

Comment:
Can this happen? According to [the definition of `TideData`](https://github.com/kubernetes/test-infra/blob/6d4635387dcce1b6f1f494b9e6312c31d4a9aecc/prow/cmd/deck/static/api/tide.ts#L57-L61) (which agrees with [the data source](https://github.com/kubernetes/test-infra/blob/6d4635387dcce1b6f1f494b9e6312c31d4a9aecc//prow/cmd/deck/tide.go#L35-L39)), it can't, in which case this check makes no sense.

If you think it can happen, then a) please explain when it happens, and then b) fix the typescript definition of `TideData` to match. You'll probably subsequently have to make a lot of other changes too.

Reply:
This fixed a null pointer error that was coming up in my test instance when there were no TideQueries. Honestly, I'm not really sure what was going on. Should I just change it back?

Reply:
This is probably a real bug that should probably be fixed. I haven't traced the code in depth, but my educated guess is as follows:

Go considers `nil` to be a valid, empty slice. Consequently, it is idiomatic to initialise a slice with `nil` (by doing e.g. `var someSlice []string` or whatever). This `nil` then eventually gets JSON serialised as `null` (because it's _not_ really an empty slice, it's just null). While the Go structs do not specify `omitempty`, and so the empty value is included, the empty value is `null` rather than an empty array (`[]`). Unfortunately, Go is unusual in thinking that null is a valid array, and so when we later parse that in TypeScript, it is `null` with no type and any attempt to operate on it fails.

To fix this, we should either change the TypeScript type so it accepts nulls (and then add null checks probably all over the place), or change the Go code so it can never be `nil` (e.g. by actually initialising to an empty slice somewhere).

In either case, though, the fix probably belongs in another PR - I'd file an issue, remove this change, and follow up later.

Reply:
> Go is unusual in thinking that null is a valid array

If we're being _very_ pedantic, `nil` is a valid (zero) value for a slice, but not an array ;)

Reply:
Added an [issue](https://github.com/kubernetes/test-infra/issues/13061)